### PR TITLE
[DEN-412] Updates, deletion, and creation of http large origins

### PR DESCRIPTION
**Pull request by ssaluja4**

### CODE REVIEW

File: edgecast/resources/originv3/origin_httplarge.go [hunk 1]
```diff
@@ -178,21 +187,147 @@ func ResourceOriginGroupUpdate(
...
-	if err != nil {
-		return diag.FromErr(err)
+	// Spin up a worker to call the api.
+	wg.Add(1)
+	go func(updateParams originv3.UpdateHttpLargeGroupParams) {
+		defer wg.Done()
+
+		_, err = svc.HttpLargeOnly.UpdateHttpLargeGroup(updateParams)
+		if err == nil {
+			mlock.Lock()
```
Comment (stevenpaz) [inline edgecast/resources/originv3/origin_httplarge.go:219]: No need to lock - log is thread safe

Comment (stevenpaz) [reply edgecast/resources/originv3/origin_httplarge.go:219]: But, keep the lock around errs append

File: edgecast/resources/originv3/origin_httplarge_test.go [hunk 2]
```diff
@@ -279,7 +266,7 @@ func TestExpandOrigins(t *testing.T) {
 		{
 			name:          "nil input",
 			input:         nil,
-			expectedPtr:   make([]*originv3.CustomerOriginRequest, 0),
+			expectedPtr:   make([]*OriginState, 0),
 			expectSuccess: false,
 		},
 	}
```
Comment (stevenpaz) [inline edgecast/resources/originv3/origin_httplarge_test.go:272]: Would be good to have a test for getOriginsDelta

**Review by stevenpaz — CHANGES_REQUESTED** [review]

File: edgecast/resources/originv3/origin_httplarge.go (hunk 1 above)
Comment (ssaluja4) [reply edgecast/resources/originv3/origin_httplarge.go]: removed locks around logs

File: edgecast/resources/originv3/origin_httplarge_test.go (hunk 2 above)
Comment (ssaluja4) [reply edgecast/resources/originv3/origin_httplarge_test.go:272]: Refactored the delta code to make it unit testable.

**Review by stevenpaz — APPROVED** [review]

**Pull request merged**